File: crates/rolldown_common/src/inner_bundler_options/types/platform.rs
#[cfg(feature = "deserialize_bundler_options")]
use schemars::JsonSchema;
#[cfg(feature = "deserialize_bundler_options")]
use serde::Deserialize;

#[derive(Debug, Clone, Copy, PartialEq, Eq)]
#[cfg_attr(feature = "deserialize_bundler_options", derive(Deserialize, JsonSchema))]
#[cfg_attr(feature = "deserialize_bundler_options", serde(rename_all = "camelCase"))]
pub enum Platform {
  /// Represents the Node.js platform.
  Node,
  Browser,
  Neutral,
  /// Represents the WASI Preview 1 platform.
  Wasi,
  /// Represents the WASI Preview 2 platform.
  WasiP2,
}

impl TryFrom<&str> for Platform {
  type Error = String;

  fn try_from(value: &str) -> Result<Self, Self::Error> {
    match value {
      "node" => Ok(Self::Node),
      "browser" => Ok(Self::Browser),
      "neutral" => Ok(Self::Neutral),
      "wasi" | "wasip1" => Ok(Self::Wasi),
      "wasip2" => Ok(Self::WasiP2),
      _ => Err(format!("Unknown platform: {value:?}")),
    }
  }
}

File: crates/rolldown_common/src/inner_bundler_options/types/platform.js
/**
 * Mock implementation of the Platform enum from Rust
 */
export const Platform = {
  Node: 0,
  Browser: 1,
  Neutral: 2,
  Wasi: 3,
  WasiP2: 4,
};

/**
 * Try to convert a string to a Platform enum value
 */
export function tryFrom(value) {
  switch (value) {
    case "node": return Platform.Node;
    case "browser": return Platform.Browser;
    case "neutral": return Platform.Neutral;
    case "wasi":
    case "wasip1": return Platform.Wasi;
    case "wasip2": return Platform.WasiP2;
    default: throw new Error(`Unknown platform: ${value}`);
  }
} 